source/common/router/router.cc

```diff
@@ -738,6 +744,12 @@ Http::FilterDataStatus Filter::decodeData(Buffer::Instance& data, bool end_strea
   }
 
   if (end_stream) {
+    Upstream::ClusterRequestResponseSizeStatsOptRef req_resp_stats =
+        cluster_->requestResponseSizeStats();
+    if (req_resp_stats.has_value()) {
+      req_resp_stats->get().upstream_rq_body_size_.recordValue(
+          callbacks_->streamInfo().bytesReceived());
+    }
```

Comment:
This only captures the full data if the body was fully buffered by another filter since decodeData can be called multiple times

Reply:
Isn't callbacks_->streamInfo().bytesReceived() meant to be cumulative?

https://github.com/envoyproxy/envoy/blob/main/include/envoy/stream_info/stream_info.h#L289

Reply:
Ok per offline chat, moving this into UpstreamRequest and using the stream_info_ there when applicable.